Library/Source/NativeEngine.cpp

```diff
@@ -421,19 +423,29 @@ namespace babylon
     {
         const auto& vertexArray = *(info[0].As<Napi::External<VertexArray>>().Data());
 
-        bgfx::setIndexBuffer(vertexArray.indexBuffer.handle);
+        if (vertexArray.indexBuffer.handle.idx != bgfx::kInvalidHandle)
+        {
+            bgfx::setIndexBuffer(vertexArray.indexBuffer.handle);
+        }
 
         const auto& vertexBuffers = vertexArray.vertexBuffers;
         for (uint8_t index = 0; index < vertexBuffers.size(); ++index)
         {
             const auto& vertexBuffer = vertexBuffers[index];
-            bgfx::setVertexBuffer(index, vertexBuffer.handle, vertexBuffer.startVertex, UINT32_MAX, vertexBuffer.declHandle);
+            if (vertexBuffer.handle.idx != bgfx::kInvalidHandle) // TODO : why does this happen?
+            {
+                bgfx::setVertexBuffer(index, vertexBuffer.handle, vertexBuffer.startVertex, UINT32_MAX, vertexBuffer.declHandle);
+            }
         }
     }
 
     Napi::Value NativeEngine::CreateIndexBuffer(const Napi::CallbackInfo& info)
     {
         const Napi::TypedArray data = info[0].As<Napi::TypedArray>();
+        if (!data.ElementLength())
```

Comment:
From the JS side, you can create an empty index array without any issue. But bgfx will assert. Someone has to do the check.

Reply:
nit: I know it means the same thing, but I greatly prefer to avoid Boolean operators for non-Boolean types. I don't know whether we've explicitly established a convention on this, but for cases like these I'd much prefer to directly check `data.ElementLength() == 0`. As a human reading the code, I just find that much more difficult to misunderstand.

Reply:
I guess I don't like using `bgfx::InvalidHandle` as a valid value that can be returned. I would prefer to do this on the JS side.